third_party/parser/codegen/diagnostics.rb

```diff
@@ -1,126 +0,0 @@
-
-  # Parser errors
-  :nth_ref_alias           => 'cannot define an alias for a back-reference variable',
-  :begin_in_method         => 'BEGIN in method',
-  :backref_assignment      => 'cannot assign to a back-reference variable',
-  :invalid_assignment      => 'cannot assign to a keyword',
-  :module_name_const       => 'class or module name must be a constant literal',
-  :unexpected_token        => 'unexpected token: {}',
-  :argument_const          => 'formal argument cannot be a constant',
-  :argument_ivar           => 'formal argument cannot be an instance variable',
-  :argument_gvar           => 'formal argument cannot be a global variable',
-  :argument_cvar           => 'formal argument cannot be a class variable',
-  :duplicate_argument      => 'duplicate argument name',
-  :empty_symbol            => 'empty symbol literal',
-  :odd_hash                => 'odd number of entries for a hash',
-  :singleton_literal       => 'cannot define a singleton method for a literal',
-  :dynamic_const           => 'dynamic constant assignment',
-  :const_reassignment      => 'constant re-assignment',
-  :module_in_def           => 'module definition in method body',
-  :class_in_def            => 'class definition in method body',
-  :unexpected_percent_str  => '{}: unknown type of percent-literal',
-  :block_and_blockarg      => 'both block argument and literal block are passed',
-  :masgn_as_condition      => 'multiple assignment in conditional context',
-  :block_given_to_yield    => 'block given to yield',
-  :invalid_regexp          => '{}',
-
-  # Parser warnings
-  :useless_else            => 'else without rescue is useless',
-
-  # Parser errors that are not Ruby errors
-  :invalid_encoding        => 'literal contains escape sequences incompatible with UTF-8',
-
-  # Rewriter diagnostics
-  :invalid_action          => 'cannot {}',
-  :clobbered               => 'clobbered by: {}',
-
-  # TypedRuby diagnostics
-  :not_static_cpath_in_geninst => "Type name in generic instance must be a static constant path"
-}.freeze
-
-def camelize(sym)
-  sym.to_s.split('_').collect(&:capitalize).join
-end
-
-def generate_rs(out)
-  out.puts "// This file is autogenerated by diagnostics.rb"
-  out.puts "// DO NOT MODIFY"
-  out.puts "#[repr(C)]"
-  out.puts "#[derive(Debug)]"
-  out.puts "#[derive(PartialEq)]"
-  out.puts "pub enum Error {"
-  MESSAGES.each do |err, _|
-    out.puts "    #{camelize(err)},"
-  end
-  out.puts "}"
-  out.puts "\n\n"
-
-  out.puts "impl fmt::Display for Diagnostic {"
-  out.puts "    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {"
-  out.puts "        let arg1 = match self.data { Some(ref str) => str.as_str(), None => \"[??]\" };"
-  out.puts "        match self.error {"
-  MESSAGES.each do |err, msg|
-    args = ", arg1" if msg.include? '{}'
-    out.puts "            Error::#{camelize(err)} => write!(f, #{msg.inspect}#{args}),"
-  end
-  out.puts "        }"
-  out.puts "    }"
-  out.puts "}"
-end
```

Comment:
I didn't keep this logic to generate rust bindings, because we've long since stopped using it.